### DISCUSSION TITLE
patcher

### DISCUSSION THREAD

**Discussion by demimike** [gruntwork-io/knowledge-base#754] [Help]
In setting up the github auth token, what access rights does the patcher require?
Does this currently work with the AWS-service-catalog? When I run it it tells me everything is up to date, even when I have different versions of the same module.
![image](https://github.com/gruntwork-io/knowledge-base/assets/98773415/8eb43505-86a0-43cb-a951-b07d07825888)

---

<ins datetime="2023-07-17T20:17:57Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110349">Tracked in ticket #110349</a></p>
</ins>

**Comment by robmorgan** [comment]
Hi @demimike, thank you for sharing your questions! I'll double-check all of these with the team. On a side note, can you provide the logs from executing that command? After running Patcher `report` or `update`, run the following to view the latest log file:

```
cat ~/.patcher/logs/"$(ls -1rt ~/.patcher/logs | tail -n1)"
```

**Reply by techpink** [reply]
Hi @demimike, when providing us logs can you please enable [debug logging](https://docs.gruntwork.io/patcher/#debugging) first:
```
patcher report --loglevel debug
```

**Reply by techpink** [reply]
> In setting up the github auth token, what access rights does the patcher require?

The token needs to have permissions to read the [`gruntwork-io`](https://github.com/gruntwork-io) private repos. Patcher checks performs a basic validation of the token, so that's unlikely to be the source of the issue.

**Comment by techpink** [comment]
> Does this currently work with the AWS-service-catalog?

Yes it does. Based on the screenshot you've provided, this is an issue we've not seen before and we'd appreciate you taking the time to reproduce the issue and send us the log file. If you'd prefer not to attach the log file here, please send it support@gruntwork.io.

**Comment by demimike** [comment]
I included a log from a token with no rights vs everything. I would 
prefer to just give it the minimum rights necessary to work.

On Tue, Jul 25 2023 at 05:42:53 AM -0700, Griff 
***@***.***> wrote:
>> Does this currently work with the AWS-service-catalog?
>> 
> Yes it does. Based on the screenshot you've provided, this is an 
> issue we've not seen before and we'd appreciate you taking the time 
> to reproduce the issue and send us the log file. If you'd prefer not 
> to attach the log file here, please send it ***@***.*** 
> ***@***.***>.
> 
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/orgs/gruntwork-io/discussions/754#discussioncomment-6540002>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AXRSTJ2XOA4VZQNGVYHD66TXR65M3ANCNFSM6AAAAAA2NNZZ4A>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
> 


## patcher log haveing token with no rights.

time=2023-07-25T08:02:29-05:00 level=info msg=Discovered 28 matching files
time=2023-07-25T08:02:29-05:00 level=info msg=Extracting dependencies from 28 files...
time=2023-07-25T08:05:39-05:00 level=info msg=Extracted 14 unique dependencies from 28 files...
time=2023-07-25T08:05:39-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:05:40-05:00 level=error msg=Could not list versions for gruntwork-io/terraform-aws-service-catalog: GET https://api.github.com/repos/gruntwork-io/terraform-aws-service-catalog/tags?page=1&per_page=100: 404 Not Found []

time=2023-07-25T08:05:40-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/route53: 0
time=2023-07-25T08:05:40-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/route53: 0
time=2023-07-25T08:05:40-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/route53: 0
time=2023-07-25T08:05:40-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:05:40-05:00 level=error msg=Could not list versions for gruntwork-io/terraform-aws-service-catalog: GET https://api.github.com/repos/gruntwork-io/terraform-aws-service-catalog/tags?page=1&per_page=100: 404 Not Found []

time=2023-07-25T08:05:40-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/vpc: 0
time=2023-07-25T08:05:40-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/vpc: 0
time=2023-07-25T08:05:40-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:05:40-05:00 level=error msg=Could not list versions for gruntwork-io/terraform-aws-service-catalog: GET https://api.github.com/repos/gruntwork-io/terraform-aws-service-catalog/tags?page=1&per_page=100: 404 Not Found []

time=2023-07-25T08:05:40-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/data-stores/redis: 0
time=2023-07-25T08:05:40-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:05:40-05:00 level=error msg=Could not list versions for demiplane-org/infrastructure-modules: GET https://api.github.com/repos/demiplane-org/infrastructure-modules/tags?page=1&per_page=100: 404 Not Found []

time=2023-07-25T08:05:40-05:00 level=info msg=Found new versions for dependency demiplane-org/infrastructure-modules/custom-iam-entity: 0
time=2023-07-25T08:05:40-05:00 level=info msg=Found new versions for dependency demiplane-org/infrastructure-modules/custom-iam-entity: 0
time=2023-07-25T08:05:40-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:05:40-05:00 level=error msg=Could not list versions for gruntwork-io/terraform-aws-service-catalog: GET https://api.github.com/repos/gruntwork-io/terraform-aws-service-catalog/tags?page=1&per_page=100: 404 Not Found []

time=2023-07-25T08:05:40-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/landingzone/account-baseline-app: 0
time=2023-07-25T08:05:40-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:05:40-05:00 level=error msg=Could not list versions for gruntwork-io/terraform-aws-service-catalog: GET https://api.github.com/repos/gruntwork-io/terraform-aws-service-catalog/tags?page=1&per_page=100: 404 Not Found []

time=2023-07-25T08:05:40-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/data-stores/aurora: 0
time=2023-07-25T08:05:40-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:05:40-05:00 level=error msg=Could not list versions for gruntwork-io/terraform-aws-service-catalog: GET https://api.github.com/repos/gruntwork-io/terraform-aws-service-catalog/tags?page=1&per_page=100: 404 Not Found []

time=2023-07-25T08:05:40-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/mgmt/ecs-deploy-runner: 0
time=2023-07-25T08:05:40-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:05:41-05:00 level=error msg=Could not list versions for gruntwork-io/terraform-aws-service-catalog: GET https://api.github.com/repos/gruntwork-io/terraform-aws-service-catalog/tags?page=1&per_page=100: 404 Not Found []

time=2023-07-25T08:05:41-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/sns-topics: 0
time=2023-07-25T08:05:41-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:05:41-05:00 level=error msg=Could not list versions for demiplane-org/infrastructure-modules: GET https://api.github.com/repos/demiplane-org/infrastructure-modules/tags?page=1&per_page=100: 404 Not Found []

time=2023-07-25T08:05:41-05:00 level=info msg=Found new versions for dependency demiplane-org/infrastructure-modules/data-stores/dynamodb: 0
time=2023-07-25T08:05:41-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:05:41-05:00 level=error msg=Could not list versions for gruntwork-io/terraform-aws-service-catalog: GET https://api.github.com/repos/gruntwork-io/terraform-aws-service-catalog/tags?page=1&per_page=100: 404 Not Found []

time=2023-07-25T08:05:41-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/mgmt/openvpn-server: 0
time=2023-07-25T08:05:41-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:05:41-05:00 level=error msg=Could not list versions for gruntwork-io/terraform-aws-vpc: GET https://api.github.com/repos/gruntwork-io/terraform-aws-vpc/tags?page=1&per_page=100: 404 Not Found []

time=2023-07-25T08:05:41-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-vpc/vpc-peering-cross-accounts-requester: 0
time=2023-07-25T08:05:41-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:05:41-05:00 level=error msg=Could not list versions for gruntwork-io/terraform-aws-service-catalog: GET https://api.github.com/repos/gruntwork-io/terraform-aws-service-catalog/tags?page=1&per_page=100: 404 Not Found []

time=2023-07-25T08:05:41-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/alb: 0
time=2023-07-25T08:05:41-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/alb: 0
time=2023-07-25T08:05:41-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/alb: 0
time=2023-07-25T08:05:41-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:05:41-05:00 level=error msg=Could not list versions for gruntwork-io/terraform-aws-service-catalog: GET https://api.github.com/repos/gruntwork-io/terraform-aws-service-catalog/tags?page=1&per_page=100: 404 Not Found []

time=2023-07-25T08:05:41-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/services/ecs-fargate-cluster: 0
time=2023-07-25T08:05:41-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/services/ecs-fargate-cluster: 0
time=2023-07-25T08:05:41-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/services/ecs-fargate-cluster: 0
time=2023-07-25T08:05:41-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:05:41-05:00 level=error msg=Could not list versions for gruntwork-io/terraform-aws-service-catalog: GET https://api.github.com/repos/gruntwork-io/terraform-aws-service-catalog/tags?page=1&per_page=100: 404 Not Found []

time=2023-07-25T08:05:41-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/vpc-mgmt: 0
time=2023-07-25T08:05:41-05:00 level=info msg=Finished extracting dependencies

## Patcher log with token having everything

time=2023-07-25T08:12:41-05:00 level=info msg=Discovered 28 matching files
time=2023-07-25T08:12:41-05:00 level=info msg=Extracting dependencies from 28 files...
time=2023-07-25T08:15:52-05:00 level=info msg=Extracted 14 unique dependencies from 28 files...
time=2023-07-25T08:15:52-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:15:53-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:15:53-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/alb: 109
time=2023-07-25T08:15:53-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/alb: 109
time=2023-07-25T08:15:53-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/alb: 1
time=2023-07-25T08:15:53-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/sns-topics: 109
time=2023-07-25T08:15:53-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/vpc-mgmt: 109
time=2023-07-25T08:15:53-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/landingzone/account-baseline-app: 109
time=2023-07-25T08:15:54-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/route53: 1
time=2023-07-25T08:15:54-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/route53: 109
time=2023-07-25T08:15:54-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/route53: 1
time=2023-07-25T08:15:54-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/data-stores/redis: 109
time=2023-07-25T08:15:54-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:15:54-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-vpc/vpc-peering-cross-accounts-requester: 15
time=2023-07-25T08:15:54-05:00 level=info msg=Calling client.Repositories.ListReleases()
time=2023-07-25T08:15:55-05:00 level=info msg=Found new versions for dependency demiplane-org/infrastructure-modules/data-stores/dynamodb: 6
time=2023-07-25T08:15:55-05:00 level=error msg=Unexpected status code 404 trying to fetch releases: https://docs.gruntwork.io/patcher-changelogs/infrastructure-modules/data-stores/dynamodb.json
time=2023-07-25T08:15:55-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/mgmt/openvpn-server: 109
time=2023-07-25T08:15:55-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/services/ecs-fargate-cluster: 1
time=2023-07-25T08:15:55-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/services/ecs-fargate-cluster: 1
time=2023-07-25T08:15:55-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/services/ecs-fargate-cluster: 1
time=2023-07-25T08:15:55-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/mgmt/ecs-deploy-runner: 109
time=2023-07-25T08:15:55-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/vpc: 109
time=2023-07-25T08:15:55-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/networking/vpc: 1
time=2023-07-25T08:15:55-05:00 level=info msg=Found new versions for dependency gruntwork-io/terraform-aws-service-catalog/data-stores/aurora: 1
time=2023-07-25T08:15:56-05:00 level=info msg=Found new versions for dependency demiplane-org/infrastructure-modules/custom-iam-entity: 0
time=2023-07-25T08:15:56-05:00 level=info msg=Found new versions for dependency demiplane-org/infrastructure-modules/custom-iam-entity: 0
time=2023-07-25T08:15:56-05:00 level=info msg=Finished extracting dependencies

**Comment by gitsstewart** [comment]
@robmorgan have you been able to assist with this further? If not, is it something that Ana could start looking at with you?
